hepcrawl/parsers/elsevier.py

```diff
@@ -0,0 +1,621 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of hepcrawl.
+# Copyright (C) 2015, 2016, 2017 CERN.
+#
+# hepcrawl is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+from __future__ import absolute_import, division, print_function
+
+import itertools
+import re
+from itertools import chain
+
+import six
+
+from inspire_schemas.api import LiteratureBuilder, ReferenceBuilder
+from inspire_schemas.utils import split_page_artid
+from inspire_utils.date import PartialDate
+from inspire_utils.helpers import maybe_int, remove_tags
+
+from ..utils import get_first, get_node
+
+
+DOCTYPE_MAPPING = {
+            'abs': 'abstract',
+            'add': 'addendum',
+            'adv': 'advertisement',
+            'ann': 'announcement',
+            'brv': 'book-review',
+            'cal': 'calendar',
+            'chp': 'chapter in a book',
+            'cnf': 'conference',
+            'con': 'contents list',
+            'cop': 'copyright information',
+            'cor': 'correspondence',
+            'crp': '',
+            'dis': 'discussion',
+            'dup': 'duplicate',
+            'edb': 'editorial board',
+            'edi': 'editorial',
+            'err': 'erratum',
+            'exm': 'exam',
+            'fla': 'full-length article',
+            'ind': 'index',
+            'lit': 'literature alert',
+            'mis': 'miscellaneous',
+            'nws': 'news',
+            'ocn': 'other contents',
+            'pgl': 'practice guidelines',
+            'pnt': 'patent report',
+            'prp': 'personal report',
+            'prv': 'product review',
+            'pub': 'publisher\'s Note',
+            'rem': 'removal',
+            'req': 'request for assistance',
+            'ret': 'retraction',
+            'rev': 'review',
+            'sco': 'short communication',
+            'ssu': 'short survey',
+        }
+
+COPYRIGHT_MAPPING = {
+    'crown': 'Crown copyright',
+    'free-of-copyright': 'None',
+    'full-transfer': 'Publisher',
+    'joint': 'Publisher and the society',
+    'limited-transfer': 'Publisher and the author',
+    'other': 'Authors',
+    'society': 'The society',
+    'unknown': 'Unknown',
+    'us-gov': ' US government'
```

Comment:
are you sure about this? I don't think it's very useful to have `Unknown` in the record, better not put anything in that case.

Reply:
That's whats written in [The Elsevier DTD 5 Family of XML DTDs](https://www.elsevier.com/__data/assets/pdf_file/0010/851626/tagbytag5-v1.9.7.pdf):
```
• crown is used when the author claims Crown copyright. 
• free-of-copyright: this value is used when the item has no copyright, for example in case of an index. 
• full-transfer: this value is used when a full transfer to one of the publisher’s
companies has been received.
• joint is used when a full transfer has been received for an article in a journal whose
copyright is owned jointly by one of the publisher’s companies and a society. 
• limited-transfer is used when the author has granted only limited rights; special
care must be taken for its production.
• no-transfer is used when there is an unresolvable copyright problem and the article may not be published (in principle, documents with this copyright status cannot
occur). This should not be confused with copyright status 001, when copyright has
not yet been transferred. Within ce:figure, ce:textbox or ce:e-component it
is used to signal that the object may not appear online.
• other is used when copyright owner is different from the journal’s copyright owner,
e.g. the authors or their employing institutions. This copyright type is also used in
the ce:copyright within a ce:figure, ce:textbox or ce:e-component. 
• society is used when a full transfer has been received for an article in a journal
whose copyright is owned by a society. 
• unknown: this value is used when the article may be published but the actual status
is unknown. This is, for instance, the case when the copyright transfer form has not
yet been received from the author. 
• us-gov is used when the author is a US government employee and will not transfer
copyright. 
```
To me it looks the most reasonable for the copyright owner in case when no other information is provided.